app/screens/channel_files/index.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {observeChannel} from '@queries/servers/channel';
+import {observeCanDownloadFiles, observeConfigBooleanValue} from '@queries/servers/system';
+import {observeCurrentTeam} from '@queries/servers/team';
+
+import ChannelFiles from './channel_files';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+type Props = WithDatabaseArgs & {
+    channelId: string;
+}
+
+const enhance = withObservables(['channelId'], ({channelId, database}: Props) => {
+    const team = observeCurrentTeam(database);
+    const channel = observeChannel(database, channelId);
+    return {
+        teamId: team.pipe(switchMap((t) => of$(t?.id))),
```

Comment:
Do we want always the current team id? Or do you foresee any case where we may be looking at a channel in a different team? If that is the case, it would make sense to get the channel team id, and default to the current team id (for DM/GMs).